diff --git a/calculator.py b/calculator.py
--- a/calculator.py
+++ b/calculator.py
@@ -25,8 +25,8 @@
             button.grid(row=row, column=col, sticky='e')
-            if text in '+-*/=':
-                button.config(bg='yellow')
-            elif text == '=':
-                button.config(bg='green')
+            if text == '=':
+                button.config(bg='green')
+            elif text in '+-*/':
+                button.config(bg='yellow')
             elif text in 'CCE':
                 button.config(bg='red')
 
diff --git a/calculator_logic.py b/calculator_logic.py
--- a/calculator_logic.py
+++ b/calculator_logic.py
@@ -8,5 +8,7 @@
     def add_digit(self, digit: int):
+        if self.display_text == 'Error':
+            self.display_text = '0'
         if self.display_text == '0':
             self.display_text = str(digit)
         else: